### DISCUSSION TITLE
How to fix application to workspace I opened it in ?

### DISCUSSION THREAD

**Discussion by yagarea** [i3/i3#4782] [Q&A] [answered]
Model situation:
\> I open my IDE in _workspace1_
\> IDE shows loading screen
\> I move to _workspace2_ to my browser to read documentation
\> IDE loads up and pops up in _workspace2_
\> Me having to move it back to _worskapce1_

Is there a way to make child window stay in workspace of parent ?

Thank you for help

**Comment by Airblader** [comment, marked as answer]
If you

* start the application using i3's `exec` command and
* do _not_ use `--no-startup-id`

… then i3 will already do this. However, the application (the IDE) needs to support the startup notification protocol. Otherwise i3 doesn't have enough information to determine this.

If it should _always_ be workspace 1 (rather than "whatever workspace you started the application on"), you can also use [`assign`](https://i3wm.org/docs/userguide.html#assign_workspace).

**Reply by yagarea** [reply]
Thank you for answer!  
Is there a way to start application with i3's `exec` from rofi ?

**Reply by aalvarado** [reply]
I use the following but not sure if this is right:

```ini
bindsym $mod+d exec --no-startup-id rofi -show run -lines 5 -run-command "i3-msg exec '{cmd}'"
```
